app/code/Magento/CustomerImportExport/Model/Import/Customer.php

```diff
@@ -270,13 +270,23 @@ protected function _saveCustomerEntities(array $entitiesToCreate, array $entitie
             $this->_connection->insertOnDuplicate(
                 $this->_entityTable,
                 $entitiesToUpdate,
-                $this->customerFields
+                $this->getCustomerEntityFieldsToUpdate($entitiesToUpdate)
             );
         }
 
         return $this;
     }
 
+    private function getCustomerEntityFieldsToUpdate(array $entitiesToUpdate): array
+    {
+        $firstCustomer = reset($entitiesToUpdate);
+        $columnsToUpdate = array_keys($firstCustomer);
+        $customerFieldsToUpdate = array_filter($this->customerFields, function ($field) use ($columnsToUpdate) {
+            return in_array($field, $columnsToUpdate);
+        });
+        return $customerFieldsToUpdate:
```

Comment:
Change `:` to `;` Please

Reply:
ups! I edited the commit. Thanks for noticing.